### PR TITLE
Fix memory leak in ChainStateWithEBBs

### PR DESCRIPTION
**Pull request by edsko**

### CODE REVIEW

File: ouroboros-consensus/src/Ouroboros/Consensus/Protocol/WithEBBs.hs [hunk 1]
```diff
@@ -1,3 +1,6 @@
+{-# LANGUAGE BangPatterns            #-}
+{-# LANGUAGE DeriveAnyClass          #-}
```
Comment (nc6) [inline ouroboros-consensus/src/Ouroboros/Consensus/Protocol/WithEBBs.hs]: Why is this needed? Or `DeriveGeneric`?

Comment (edsko) [reply ouroboros-consensus/src/Ouroboros/Consensus/Protocol/WithEBBs.hs]: It's not yet, but it will in my next PR :) Remnant from when I rebased this on `master`. But we'll need it as soon as I submit the next (big) PR.
